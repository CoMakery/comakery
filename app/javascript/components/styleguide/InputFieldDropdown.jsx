import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class InputFieldDropdown extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'input-field__dropdown',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="select"
          symbolLimit={0}
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default InputFieldDropdown
