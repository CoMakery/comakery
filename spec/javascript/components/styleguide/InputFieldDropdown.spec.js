import React from 'react'
import { shallow, mount, render } from 'enzyme'
import InputFieldDropdown from 'components/styleguide/InputFieldDropdown'

describe('InputFieldDropdown', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<InputFieldDropdown/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.input-field__dropdown').props().type).toBe('select')
      expect(wrapper.exists('.input-field--title--counter')).not.toBe()
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<InputFieldDropdown className='__test' />)

      expect(wrapper.exists('.input-field__dropdown.__test')).toBe(true)
    })
})
