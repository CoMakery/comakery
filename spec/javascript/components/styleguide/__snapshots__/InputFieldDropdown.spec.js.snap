// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldDropdown renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="input-field__dropdown"
    disabled={false}
    errorText=""
    eventHandler={[Function]}
    imgPreviewDimensions="40x40"
    imgPreviewUrl=""
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    required={false}
    selectEntries={Array []}
    symbolLimit={0}
    title="title"
    type="select"
    value=""
  />
</Fragment>
`;
